lib/instrumentation/http-shared.js

```diff
@@ -191,24 +192,22 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
 
       return req
 
-      function onresponse (res) {
-        // Work around async_hooks bug in Node.js 12.0 - 12.2 (https://github.com/nodejs/node/pull/27477)
-        ins._recoverTransaction(span.transaction)
-
-        agent.logger.debug('intercepted http.ClientRequest response event %o', { id: id })
-        ins.bindEmitter(res)
-
-        res.prependListener('end', onEnd)
+      // In case the request is ended prematurely
+      function onAbort (type) {
```

Comment:
@axw I changed the original implementation which relied on end of steam package and used our previous emit logic for handling the `abort` events, its easier and does not add event listeners for unnecessary events.